src/interfaces/wallet.cpp

```diff
@@ -60,16 +60,16 @@ WalletTx MakeWalletTx(CWallet& wallet, const CWalletTx& wtx)
 }
 
 //! Construct wallet tx status struct.
-WalletTxStatus MakeWalletTxStatus(interfaces::Chain::Lock& locked_chain, const CWalletTx& wtx)
+WalletTxStatus MakeWalletTxStatus(CWallet& wallet, const CWalletTx& wtx)
 {
     WalletTxStatus result;
-    result.block_height = locked_chain.getBlockHeight(wtx.m_confirm.hashBlock).get_value_or(std::numeric_limits<int>::max());
+    result.block_height = wtx.m_confirm.block_height;
```

Comment:
Is behavior changing here? Does this need need to be changed to continue returning `std::numeric_limits<int>::max()` in the case of an abandoned or unconfirmed transaction, or does it not matter?

Reply:
I think it doesn't matter, and maybe even confusing to display `std::numeric_limit<int>::max()` as block height. An abandoned one will have `is_abandoned` sets to true. An unconfirmed one will have `depth_in_main_chain` sets to 0. It seems to me meaningful enough for end user but I can still do a ternary and if `hashBlock` is null set it to fallback value if you think so.

Reply:
> I think it doesn't matter, and maybe even confusing to display `std::numeric_limit<int>::max()` as block height. An abandoned one will have `is_abandoned` sets to true. An unconfirmed one will have `depth_in_main_chain` sets to 0. It seems to me meaningful enough for end user but I can still do a ternary and if `hashBlock` is null set it to fallback value if you think so.

I think anything that doesn't break the GUI is fine. If GUI treats 0x7FFFFFFFF and 0xFFFFFFFFF the same, the current code is fine. If GUI treats them differently, that should just be addressed somehow. I'm just asking because I'm not sure if this is a real change or not.

Reply:
Back to `result.block_height = wtx.m_confirm.block_height > 0 ? wtx.m_confirm.block_height std::numeric_limits<int>::max();`

Should avoid to break qt code, without further digging into it.